src/loki/worker.cc

```diff
@@ -83,13 +84,35 @@ void loki_worker_t::parse_costing(Api& api, bool allow_none) {
     }
   } catch (const std::runtime_error&) { throw valhalla_exception_t{125, "'" + costing_str + "'"}; }
 
-  // See if we have avoids and take care of them
-  if (static_cast<size_t>(options.avoid_locations_size()) > max_avoid_locations) {
-    throw valhalla_exception_t{157, std::to_string(max_avoid_locations)};
+  if (options.avoid_polygons_size()) {
+    double rings_length = 0;
+    std::vector<ring_bg_t> rings;
+    for (const auto& ring_pbf : options.avoid_polygons()) {
+      const ring_bg_t ring = PBFToRing(ring_pbf);
+      rings.push_back(ring);
```

Comment:
this avoids any copies

```suggestion
      rings.push_back(PBFToRing(ring_pbf));
      const ring_bg_t& ring = rings.back();
```